mule-shaders-module/src/main/java/org/mule/module/shaders/api/ReflectiveAccessShader.java

```diff
@@ -0,0 +1,641 @@
+          final String filename = mappedSource.substring(mappedSource.lastIndexOf('/') + 1);
+          super.visitSource(filename, debug);
+        }
+      }
+    };
+
+    try {
+      cr.accept(cv, ClassReader.EXPAND_FRAMES);
+    } catch (Throwable ise) {
+      throw new MojoExecutionException("Error in ASM processing class " + name, ise);
+    }
+
+    byte[] renamedClass = cw.toByteArray();
+
+    // Need to take the .class off for remapping evaluation
+    String mappedName = remapper.map(name.substring(0, name.indexOf('.')));
+
+    try {
+      // Now we put it back on so the class file is written out with the right extension.
+      JarEntry entry = new JarEntry(mappedName + ".class");
+      entry.setTime(time);
+      jos.putNextEntry(entry);
+
+      jos.write(renamedClass);
+    } catch (ZipException e) {
+      getLogger().debug("We have a duplicate " + mappedName + " in " + jar);
+    }
+  }
+
+  private boolean isFiltered(List<Filter> filters, String name) {
+    for (Filter filter : filters) {
+      if (filter.isFiltered(name)) {
+        return true;
+      }
+    }
+
+    return false;
+  }
+
+  private boolean resourceTransformed(List<ResourceTransformer> resourceTransformers, String name, InputStream is,
+                                      List<Relocator> relocators, long time)
+      throws IOException {
+    boolean resourceTransformed = false;
+
+    for (ResourceTransformer transformer : resourceTransformers) {
+      if (transformer.canTransformResource(name)) {
+        getLogger().debug("Transforming " + name + " using " + transformer.getClass().getName());
+
+        if (transformer instanceof ReproducibleResourceTransformer) {
+          ((ReproducibleResourceTransformer) transformer).processResource(name, is, relocators, time);
+        } else {
+          transformer.processResource(name, is, relocators);
+        }
+
+        resourceTransformed = true;
+
+        break;
+      }
+    }
+    return resourceTransformed;
+  }
+
+  private void addJavaSource(Set<String> resources, JarOutputStream jos, String name, long time, InputStream is,
+                             List<Relocator> relocators)
+      throws IOException {
+    JarEntry entry = new JarEntry(name);
+    entry.setTime(time);
+    jos.putNextEntry(entry);
+
+    String sourceContent = IOUtil.toString(new InputStreamReader(is, StandardCharsets.UTF_8));
+
+    for (Relocator relocator : relocators) {
+      sourceContent = relocator.applyToSourceContent(sourceContent);
+    }
+
+    final Writer writer = new OutputStreamWriter(jos, StandardCharsets.UTF_8);
+    writer.write(sourceContent);
+    writer.flush();
+
+    resources.add(name);
+  }
+
+  private void addResource(Set<String> resources, JarOutputStream jos, String name, JarEntry originalEntry,
+                           JarFile jarFile)
+      throws IOException {
+    ZipHeaderPeekInputStream inputStream = new ZipHeaderPeekInputStream(jarFile.getInputStream(originalEntry));
+    try {
```

Comment:
As the class javadoc points out, `ReflectiveAccessShader` is basically a custom version of `org.apache.maven.plugins.shade.DefaultShader`, and it was copied over from that code because we needed to modify one line of code in order to inject our own class remapper, see discussions on  https://github.com/mulesoft/mule-json-module/pull/196.

While we might change to code to make it cleaner, there is no functional reason to change it because this is how the default shade plugin behaves anyway, and the next time we have to make a version upgrade for that plugin, any changes we do would just make that process more complicated.